scalameta/trees/shared/src/main/scala/scala/meta/Trees.scala

```diff
@@ -15,7 +15,7 @@ import scala.{meta => sm}
   def children: List[Tree]
 
   def pos: Position
-  def tokens(implicit dialect: Dialect): Tokens
```

Comment:
meaning, still keep the parameter but throw it origin doesn't have it?

what issues could be caused, btw?

Reply:
If we have a library depending on a certain version of Scalameta (for example Ammonite) and someone brings in the newest one then in case someone uses `tokens` method it will throw since the earlier version of the method doesn't exist.

Reply:
but then if we're planning to increment the minor version of scalameta, aren't we allowed to make binary-incompatible changes as long as they are source-compatible? wouldn't that hypothetical library be expected to be updated for newer scalameta? there must be some process to make these changes occasionally.

Reply:
Interesting enough I can't find anything about scalameta usage in Ammonite anymore, so maybe changing tokens would ok?

Anyway, I am just being super cautious here, might not be worth the trouble to keep tokens.

Reply:
for now, did you what you suggested. if you find that dropping the parameter is also ok, we'll include that in a subsequent pr.

didn't understand your point about "not worth the trouble to keep tokens".

Reply:
Thanks for checking!

> not worth the trouble to keep tokens

Ignore me, I hadn't had too much sleep at that point in time :sweat_smile: